File: Src/Icm.Core/Collections extensions/LinkedListExtensions.cs
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace Icm.Collections
{
	public static class LinkedListExtensions
	{

		/// <summary>
		/// Copy operation for linked lists.
		/// </summary>
		/// <typeparam name="T"></typeparam>
		/// <param name="l1"></param>
		/// <param name="l2"></param>
		/// <remarks></remarks>
		public static void CopyInto<T>(this LinkedList<T> l1, LinkedList<T> l2) where T : ICloneable
		{
		    var itNode = l1.First;
		    while (itNode != null) {
				T copia = (T)itNode.Value.Clone();
				l2.AddLast(copia);
				itNode = itNode.Next;
			}
		}
	}
}

File: Src2/Icm.Core/Collections extensions/LinkedListExtensions.vb
Imports System.Runtime.CompilerServices

Namespace Icm.Collections
    Public Module LinkedListExtensions

        ''' <summary>
        ''' Copy operation for linked lists.
        ''' </summary>
        ''' <typeparam name="T"></typeparam>
        ''' <param name="l1"></param>
        ''' <param name="l2"></param>
        ''' <remarks></remarks>
        <Extension()>
        Public Sub CopyInto(Of T As ICloneable)(ByVal l1 As LinkedList(Of T), ByVal l2 As LinkedList(Of T))
            Dim itNode As LinkedListNode(Of T)

            itNode = l1.First
            Do Until itNode Is Nothing
                Dim copia As T = DirectCast(itNode.Value.Clone, T)
                l2.AddLast(copia)
                itNode = itNode.Next
            Loop
        End Sub

    End Module
End Namespace
